vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,334 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdExternalNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdExternalNetworkCreate,
+		Delete: resourceVcdExternalNetworkDelete,
+		Read:   resourceVcdExternalNetworkRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_scope": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A list of IP scopes for the network",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"is_inherited": &schema.Schema{
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "True if the IP scope is inherit from parent network",
+						},
+						"gateway": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Gateway of the network",
+							ValidateFunc: validation.SingleIP(),
+						},
+						"netmask": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Network mask",
+							ValidateFunc: validation.SingleIP(),
+						},
+						"dns1": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Primary DNS server",
+							ValidateFunc: validation.SingleIP(),
+						},
+						"dns2": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
```

Comment:
Why is this property mandatory? In the GUI it is optional

Reply:
Changed